src/Parser/LAZParser.js

```diff
@@ -0,0 +1,199 @@
+import * as copc from 'copc';
+import * as THREE from 'three';
+
+const CDN_URL = 'https://unpkg.com/laz-perf@0.0.6/lib/';
+
+function parseView(view, options = { in: { colorDepth: 16 } }) {
+    const getPosition = ['X', 'Y', 'Z'].map(view.getter);
+    const getIntensity = view.getter('Intensity');
+    const getReturnNumber = view.getter('ReturnNumber');
+    const getNumberOfReturns = view.getter('NumberOfReturns');
+    const getClassification = view.getter('Classification');
+    const getPointSourceID = view.getter('PointSourceId');
+    const getColor = view.dimensions.Red ?
+        ['Red', 'Green', 'Blue'].map(view.getter) : undefined;
+
+    const positions = new Float32Array(view.pointCount * 3);
+    const intensities = new Uint16Array(view.pointCount);
+    const returnNumbers = new Uint8Array(view.pointCount);
+    const numberOfReturns = new Uint8Array(view.pointCount);
+    const classifications = new Uint8Array(view.pointCount);
+    const pointSourceIDs = new Uint16Array(view.pointCount);
+    const colors = getColor ? new Uint8Array(view.pointCount * 4) : undefined;
+
+    for (let i = 0; i < view.pointCount; i++) {
+        // `getPosition` apply scale and offset transform to the X, Y, Z values.
+        // See https://github.com/connormanning/copc.js/blob/master/src/las/extractor.ts.
+        const [x, y, z] = getPosition.map(f => f(i));
+        positions[i * 3] = x;
+        positions[i * 3 + 1] = y;
+        positions[i * 3 + 2] = z;
+
+        const intensity = getIntensity(i);
+        intensities[i] = intensity;
+
+        const returnNumber = getReturnNumber(i);
+        const numberOfReturn = getNumberOfReturns(i);
+        returnNumbers[i] = returnNumber;
+        numberOfReturns[i] = numberOfReturn;
+
+        if (getColor) {
+            // Note that we do not infer color depth as it is expensive (i.e.
+            // traverse the whole view to check if there exists a red, green or
+            // blue value > 255).
+            let [r, g, b] = getColor.map(f => f(i));
+
+            if (options.in.colorDepth === 16) {
+                r /= 256;
+                g /= 256;
+                b /= 256;
+            }
+
+            colors[i * 4] = r;
+            colors[i * 4 + 1] = g;
+            colors[i * 4 + 2] = b;
+            colors[i * 4 + 3] = 255;
+        }
+
+        const classication = getClassification(i);
+        classifications[i] = classication;
+
+        const pointSourceID = getPointSourceID(i);
+        pointSourceIDs[i] = pointSourceID;
+    }
+
+    return {
+        position: positions,
+        intensity: intensities,
+        returnNumber: returnNumbers,
+        numberOfReturns,
+        classification: classifications,
+        pointSourceID: pointSourceIDs,
+        color: colors,
+    };
+}
+
+function parseBuffer(pointData, options) {
+    const { header, eb } = options.in;
+    const view = copc.Las.View.create(pointData, header, eb);
+    const attrs = parseView(view);
```

Comment:
Whoops... Forget to pass an the `options` parameter.
The `parseView` function is only called once because I may in another PR add a `parseFile` function to parse an entire LAS/LAZ file and maybe replace the `LASParser` file.